### ISSUE TITLE
vectorize calculation of target_end_date

### ISSUE THREAD

**Issue by elray1** [reichlab/covidHubUtils#19]
Rewrite `calc_target_end_date` to accept vector inputs, then replace the calculation of target_end_date at
https://github.com/reichlab/covidHubUtils/blob/9e189a0cd209c516479d8883f4cf050ee6ae5df4/R/load_forecasts_repo.R#L117
with
`target_end_date = calc_target_end_date(forecast_date, as.numeric(horizon), target_unit)`

To vectorize `calc_target_end_date`, replace `if ... else ...` at https://github.com/reichlab/covidHubUtils/blob/9e189a0cd209c516479d8883f4cf050ee6ae5df4/R/date_management.R#L86 with `ifelse`, or possibly allocate vector of results of appropriate length and set values at indices where `target_unit == 'wk'`, then values at indices where `target_unit == 'day'`

**Comment by Serena-Wang** [comment]
Pushed the update to master. Now date_management can calculate target end dates for both weekly and daily targets. `load_forecasts` is faster now.

**Comment by nickreich** [comment]
Something still doesn't quite work here for me. See this query:
```
forecasts1 <- covidHubUtils::load_forecasts(
    models = c("COVIDhub-ensemble"),
    targets = paste(1:4, "wk ahead inc death"),
    last_forecast_date = as.Date("2020-10-05"),
    forecast_date_window_size = 3,
    source = "zoltar",
    types = "quantile")
```

yielding this error:
```
 Error: Problem with `mutate()` input `target_end_date`.
x week_offset must be either length 1 or the same length as forecast_date
ℹ Input `target_end_date` is `as.Date(...)`.
Run `rlang::last_error()` to see where the error occurred. 
```

**Comment by elray1** [comment]
New issues introduced with my previous commit are now resolved by 4ae554b.
